kimchi/src/circuits/expr.rs

```diff
@@ -520,56 +545,68 @@ impl<F: FftField> PolishToken<F> {
         let mut stack = vec![];
         let mut cache: Vec<F> = vec![];
 
+        let mut skip_count = 0;
+
         for t in toks.iter() {
-            use PolishToken::*;
-            match t {
-                Alpha => stack.push(c.alpha),
-                Beta => stack.push(c.beta),
-                Gamma => stack.push(c.gamma),
-                JointCombiner => {
-                    stack.push(c.joint_combiner.expect("no joint lookup was expected"))
-                }
-                EndoCoefficient => stack.push(c.endo_coefficient),
-                Mds { row, col } => stack.push(c.mds[*row][*col]),
-                ForeignFieldModulus(i) => {
-                    if let Some(modulus) = c.foreign_field_modulus.clone() {
-                        stack.push(ForeignElement::<F, 3>::from_biguint(modulus.clone())[*i])
+            if skip_count > 0 {
+                skip_count -= 1;
```

Comment:
to avoid the indentation of the `else { ... }` you should just write `continue;` here